src/schedule.rs

```diff
@@ -348,6 +357,8 @@ where
 {
     schedule: &'a Schedule,
     previous_datetime: Option<DateTime<Z>>,
+    later_datetime: Option<DateTime<Z>>,
+    earlier_datetime: Option<DateTime<Z>>,
```

Comment:
Since we're iterating and could receive two times at once, we need to be able to store the next result (because iterators need to return one value at a time).  They are stored separately here to allow both forward and reverse iteration.